### PR TITLE
[GEP-18] Adapt `ssh-keypair` to new secrets manager

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area security
/kind enhancement
/merge squash

**What this PR does / why we need it**:
This PR adapts the `ssh-keypair` secrets to new secrets manager. Given that there is not a single responsible component for it, the `Botanist` generates it.

The `ShootStatus` is extended with information about the SSH keypair rotation (similar to how it looks like for CA rotation). This is a requirement for the secrets manager since it needs to remember the last started time of the rotation process, and it's also an improvement for users and operators to see when the rotation was last started and/or completed.

**Which issue(s) this PR fixes**:
Part of https://github.com/gardener/gardener/issues/3292#issuecomment-1036058890

**Special notes for your reviewer**:
✅ Depends on #5524 which needs to be merged first.

/invite @timebertt @BeckerMax 

**Release note**:

```noteworthy user
There is a new section in the `ShootStatus` under `.status.credentials.rotation.sshKeypair` describing when the SSH keypair rotation was last initiated and last completed.
```

### CODE REVIEW

**Comment by ary1992** [comment]
Thank you for the well organised PR.

File: pkg/utils/secrets/manager/interface.go [hunk 1]
```diff
@@ -31,7 +31,7 @@ type Interface interface {
 	// Get returns the secret object for the secret with the given name. By default, the bundle secret will be returned.
 	// If there is no bundle secret then it falls back to the current secret. Note that only those secrets are known
 	// which were detected or generated by prior Generate calls.
-	Get(string, ...GetOption) (*corev1.Secret, error)
```
Comment (timebertt) [inline pkg/utils/secrets/manager/interface.go:34]: Can you adapt `docs/development/secrets_management.md` as well?

File: pkg/operation/botanist/secrets.go [hunk 2]
```diff
@@ -100,13 +100,16 @@ func (b *Botanist) InitializeSecretsManagement(ctx context.Context) error {
...
+		b.generateCertificateAuthorities,
+		b.generateGenericTokenKubeconfig,
+	} {
+		if err := generateFn(ctx); err != nil {
 			return err
 		}
 	}
 
-	return b.getOrGenerateGenericTokenKubeconfig(ctx)
+	return nil
```
Comment (timebertt) [inline pkg/operation/botanist/secrets.go]: If you want, you can beautify it even further: 😉 
```go
	return flow.Sequential(
		b.generateCertificateAuthorities,
		b.generateGenericTokenKubeconfig,
	)(ctx)
```

File: pkg/utils/secrets/manager/generate.go [hunk 3]
```diff
@@ -159,6 +161,57 @@ func (m *manager) keepExistingSecretsIfNeeded(ctx context.Context, configName st
...
+
+			// Wait until cache reflects changes to prevent losing the old secret.
+			timeoutCtx, cancel := context.WithTimeout(ctx, time.Minute)
+			defer cancel()
+
+			if err := retry.Until(timeoutCtx, time.Second, func(ctx context.Context) (done bool, err error) {
+				secretList := &corev1.SecretList{}
+				if err := m.client.List(ctx, secretList, client.InNamespace(m.namespace), client.MatchingLabels{
+					LabelKeyName:      configName,
+					LabelKeyManagedBy: LabelValueSecretsManager,
```
Comment (timebertt) [inline pkg/utils/secrets/manager/generate.go]: We need the identity selector here. Either has to be added here or in https://github.com/gardener/gardener/pull/5607

File: pkg/operation/botanist/component/kubeapiserver/kube_apiserver_test.go [hunk 4]
```diff
@@ -56,6 +57,18 @@ import (
...
+
+	_ = BeforeSuite(func() {
+		oldGenerateRandomString = secretutils.GenerateRandomString
+		secretutils.GenerateRandomString = secretutils.FakeGenerateRandomString
+	})
+	_ = AfterSuite(func() {
+		secretutils.GenerateRandomString = oldGenerateRandomString
+	})
+)
+
```
Comment (timebertt) [inline pkg/operation/botanist/component/kubeapiserver/kube_apiserver_test.go]: Thanks. I was already tempted to speed this up my PR as well 😄 

You could simplify this into 
```go
var _ = BeforeSuite(func() {
	DeferCleanup(test.WithVar(&secretutils.GenerateRandomString, secretutils.FakeGenerateRandomString))
})
```
(ref https://onsi.github.io/ginkgo/#cleaning-up-our-cleanup-code-defercleanup)

**Review by timebertt — COMMENTED** [review]
Awesome! Again, very well-structured and easy to review, thanks!

**Review by timebertt — APPROVED** [review]
Awesome, let's go!
/lgtm

**Pull request merged**